src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -432,15 +481,17 @@ namespace AppInstaller::CLI::Workflow
         context <<
             GetInstallerArgs <<
             RenameDownloadedInstaller <<
-            ShellExecuteInstallImpl;
+            ShellExecuteInstallImpl <<
+            ReportInstallerResult("ShellExecute"sv, APPINSTALLER_CLI_ERROR_SHELLEXEC_INSTALL_FAILED);
     }
 
     void DirectMSIInstall(Execution::Context& context)
     {
         context <<
             GetInstallerArgs <<
             RenameDownloadedInstaller <<
-            DirectMSIInstallImpl;
+            DirectMSIInstallImpl <<
+            ReportInstallerResult("MsiInstallProduct"sv, APPINSTALLER_CLI_ERROR_MSI_INSTALL_FAILED, /* isHResult */ true);

```

Comment:
>/* isHResult */ true [](http://example.com/codeflow?start=100&length=20)

MsiInstallProduct returns win32 error codes(like 1603), not hresults. So this should be false.